vcd/resource_vcd_external_network.go

```diff
@@ -0,0 +1,381 @@
+		}
+
+		vCenterHref, err := getVcenterHref(vcdClient.VCDClient, portGroup["vcenter"].(string))
+		if err != nil {
+			return &types.ExternalNetwork{}, fmt.Errorf("unable to find vCenter %s (%#v)", portGroup["vcenter"].(string), err)
+		}
+		portGroupConfiguration.VimServerRef = &types.Reference{
+			HREF: fmt.Sprintf("%s/admin/extension/vimServer/%s", vcdClient.Client.VCDHREF.String(), vCenterHref),
+		}
+
+		var portGroupRecord []*types.PortGroupRecordType
+		if portGroup["type"].(string) == "NETWORK" {
+			portGroupRecord, err = govcd.QueryNetworkPortGroups(vcdClient.VCDClient, portGroup["vsphere_network"].(string))
+		} else {
+			portGroupRecord, err = govcd.QueryDistributedPortGroups(vcdClient.VCDClient, portGroup["vsphere_network"].(string))
+		}
+		if err != nil {
+			return &types.ExternalNetwork{},
+				fmt.Errorf("unable to find port group %s (%#v)", portGroup["vcenter"].(string), err)
+		}
+
+		if len(portGroupRecord) == 1 {
+			portGroupConfiguration.MoRef = portGroupRecord[0].MoRef
+		} else {
+			return &types.ExternalNetwork{},
+				fmt.Errorf("find less or more than one port groups - %d", len(portGroupRecord))
+		}
+
+		portGroups = append(portGroups, portGroupConfiguration)
+	}
+	params.VimPortGroupRefs = &types.VimObjectRefs{
+		VimObjectRef: portGroups,
+	}
+
+	if parentNetworkName, ok := d.GetOk("parent_network"); ok {
+		name := parentNetworkName.(string)
+		parentNetwork, err := govcd.GetExternalNetworkByName(vcdClient.VCDClient, name)
+		if err != nil {
+			return &types.ExternalNetwork{}, fmt.Errorf("unable to find parent network %s (%s)", name, err)
+		}
+		params.Configuration.ParentNetwork = &types.Reference{
+			HREF: parentNetwork.HREF,
+			Type: parentNetwork.Type,
+			Name: parentNetwork.Name,
+		}
+	}
+
+	if description, ok := d.GetOk("description"); ok {
+		params.Description = description.(string)
+	}
+
+	return params, nil
+}
+
+func getVcenterHref(vcdClient *govcd.VCDClient, name string) (string, error) {
+	virtualCenters, err := govcd.QueryVirtualCenters(vcdClient, fmt.Sprintf("(name==%s)", name))
+	if err != nil {
+		return "", err
+	}
+	if len(virtualCenters) == 0 {
+		return "", fmt.Errorf("No vSphere server found with name '%s'", name)
+	}
+	return virtualCenters[0].HREF, nil
+}
+
+// Remove this function when the version of Terraform is updated; it was added in 0.11.6
+// singleIP returns a SchemaValidateFunc which tests if the provided value
+// is of type string, and in valid single IP notation
+func singleIP() schema.SchemaValidateFunc {
+	return func(i interface{}, k string) (s []string, es []error) {
+		v, ok := i.(string)
+		if !ok {
+			es = append(es, fmt.Errorf("expected type of %s to be string", k))
+			return
+		}
+
+		ip := net.ParseIP(v)
+		if ip == nil {
+			es = append(es, fmt.Errorf(
+				"expected %s to contain a valid IP, got: %s", k, v))
+		}
+		return
+	}
+}
+
+func validateFenceMode(val interface{}, key string) (warns []string, errs []error) {
```

Comment:
No need for custom function. Just use validation.StringInSlice

Reply:
fixed